waymo_raw_data_to_argoverse.py

```diff
@@ -30,205 +40,398 @@
+                timestamp_s = tf_cam_image.pose_timestamp
+                timestamp_ns = int(timestamp_s * 1e9)  # to nanoseconds
+                if args.save_poses:
+                    dump_pose(city_SE3_egovehicle, timestamp_ns, log_id, ARGO_WRITE_DIR)
+
+                if args.save_images:
+                    camera_name = CAMERA_NAMES[tf_cam_image.name]
+                    img = tf.image.decode_jpeg(tf_cam_image.image)
+                    new_img = undistort_image(
+                        np.asarray(img),
+                        frame.context.camera_calibrations,
+                        tf_cam_image.name,
+                    )
+                    img_save_fpath = f"{ARGO_WRITE_DIR}/{log_id}/{camera_name}/{camera_name}_{timestamp_ns}.jpg"
+                    # assert not Path(img_save_fpath).exists()
+                    check_mkdir(str(Path(img_save_fpath).parent))
+                    imageio.imwrite(img_save_fpath, new_img)
+                    img_count += 1
+                    if img_count % 100 == 0:
+                        print(f"\tSaved {img_count}'th image for log = {log_id}")
+
+
+def undistort_image(img: np.ndarray, calib_data: Any, camera_name: int):
+    """Undistort the image from the Waymo dataset given camera calibration data"""
+    for camera_calib in calib_data:
+        if camera_calib.name == camera_name:
+            f_u, f_v, c_u, c_v, k1, k2, p1, p2, k3 = camera_calib.intrinsic
+            camera_matrix = np.array([[f_u, 0, c_u], [0, f_v, c_v], [0, 0, 1]])
+            dist_coeffs = np.array([k1, k2, p1, p2, k3])
+            return cv2.undistort(img, camera_matrix, dist_coeffs)
+
+
+def form_calibration_json(calib_data):
+    """
+    Argoverse expects to receive "egovehicle_T_camera", i.e. from camera -> egovehicle, with
+            rotation parameterized as quaternion.
+    Waymo provides the same SE(3) transformation, but with rotation parmaeterized as 3x3 matrix
+    """
+    calib_dict = {"camera_data_": []}
+    for camera_calib in calib_data:
+        cam_name = CAMERA_NAMES[camera_calib.name]
+        # They provide "Camera frame to vehicle frame."
+        # https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/dataset.proto
+        egovehicle_SE3_waymocam = np.array(camera_calib.extrinsic.transform).reshape(
+            4, 4
+        )
+        standardcam_R_waymocam = rotY(-90).dot(rotX(90))
+        standardcam_SE3_waymocam = SE3(
+            rotation=standardcam_R_waymocam, translation=np.zeros(3)
+        )
+        waymocam_SE3_egovehicle = SE3(
+            rotation=egovehicle_SE3_waymocam[:3, :3],
+            translation=egovehicle_SE3_waymocam[:3, 3],
+        )
+        standardcam_SE3_egovehicle = standardcam_SE3_waymocam.right_multiply_with_se3(
+            waymocam_SE3_egovehicle
+        )
+        egovehicle_q_camera = rotmat2quat(standardcam_SE3_egovehicle.rotation)
+        x, y, z = standardcam_SE3_egovehicle.translation
```

Comment:
need to take x,y,z from the inverse, since on disk we are saving `"vehicle_SE3_camera_"`, not `camera_SE3_vehicle` (see line below)

Reply:
Addressed in [this commit](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/5aa62285b6622c6372a097bc05a24214f4f59757)